x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java

```diff
@@ -2089,7 +2065,7 @@ public void testForkWithStatsInAllBranches() {
                    (EVAL z = a * b | STATS m = max(z))
                    (STATS x = count(*), y=min(z))
             | WHERE x > y
-            """, Set.of("a", "a.*", "b", "b.*", "c", "c.*", "z", "z.*"));
+            """, Set.of("a", "a.*", "c", "c.*", "z", "z.*"));
```

Comment:
we correctly detect that now that we don't need to ask for `b` because it's the result of an eval